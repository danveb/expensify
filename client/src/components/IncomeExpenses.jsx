import { useContext } from "react";
import { GlobalContext } from "../context/GlobalState";

const IncomeExpenses = () => {
    // useContext
    const context = useContext(GlobalContext); 
    // destructure transactions from context
    const { transactions } = context; 

    const amounts = transactions.map((transaction) => transaction.amount); 
    const income = amounts
                    .filter((item) => item > 0)
                    .reduce((acc, item) => (acc += item), 0)
    const incomeRounded = Math.round(income); 
    const expense = amounts
                    .filter((item) => item < 0)
                    .reduce((acc, item) => (acc += item),  0) * -1
    const expenseRounded = Math.round(expense); 

    return (
        <div className="inc-exp-container">
            <div>
                <h4>Income</h4>
                <p className="money plus">$ {incomeRounded}</p>
            </div>
            <div>
                <h4>Expense</h4>
                <p className="money minus">$ {expenseRounded}</p>
            </div>
        </div>
    )
}

export default IncomeExpenses