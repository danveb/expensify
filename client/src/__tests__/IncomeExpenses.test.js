import { render } from "@testing-library/react";
import IncomeExpenses from "../components/IncomeExpenses"; 

describe("IncomeExpenses component", () => {
    // smoke test
    test("renders IncomeExpenses component", () => {
        render(<IncomeExpenses />); 
    });

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(<IncomeExpenses />); 
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByText
    test("should display income text", () => {
        const { getByText } = render(<IncomeExpenses />); 
        const text = getByText("Income", { exact: true }); 
        expect(text).toBeInTheDocument(); 
    });

    // getByText
    test("should display expense text", () => {
        const { getByText } = render(<IncomeExpenses />); 
        const text = getByText("Expense", { exact: true }); 
        expect(text).toBeInTheDocument(); 
    });
});