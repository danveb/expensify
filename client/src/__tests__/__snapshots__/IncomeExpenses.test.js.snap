// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IncomeExpenses component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="inc-exp-container"
  >
    <div>
      <h4>
        Income
      </h4>
      <p
        class="money plus"
      >
        $ 7000
      </p>
    </div>
    <div>
      <h4>
        Expense
      </h4>
      <p
        class="money minus"
      >
        $ 4650
      </p>
    </div>
  </div>
</DocumentFragment>
`;
